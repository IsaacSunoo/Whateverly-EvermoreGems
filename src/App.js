import React, { Component } from 'react';
import SplashPage from './components/SplashPage';
import GemsPage from './components/GemsPage';
import { gems, metals } from './dataset/evermoreGems';
import './App.css';

class App extends Component {
  state = {
    gems,
    metals
  }
  render() {
    return (
      <div>
        <SplashPage />
        <GemsPage />
      </div>
    );
  }
}

export default App;
