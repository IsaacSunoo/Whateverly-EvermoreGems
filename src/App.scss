* {
  box-sizing: border-box;
  font-family: "Open Sans", sans-serif;
  margin: 0 auto 0 auto;
  padding: 0;
}

body {
  background-image: url("./images/sky.jpg");
  background-repeat: no-repeat;
  background-size: 100%;
  line-height: 1.4;
}

