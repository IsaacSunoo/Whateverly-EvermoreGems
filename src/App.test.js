import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

const mockGem = [{
    "name": "Aquamarine",
    "id": 1
  },
  {
    "name": "Bixbite",
    "id": 2,
  }
]

const mockMetal = [{
    "name": "Gold",
    "id": 1
  },
  {
    "name": "Silver",
    "id": 2,
  }
]
  const mockInitalState = {
      gems: [],
      metals: [],
      selectedGem: {},
      selectedMetal: {},
      display: 'gems'
  }

describe('App initial state', () => {
  let wrapper;
  wrapper = shallow(<App />);
   it('should match the snapshot with initial state', () => {
    expect(wrapper).toMatchSnapshot();
  });
   it('should have the correct default state', () => {
    expect(wrapper.state()).toEqual(mockInitalState)
   })
})


describe('App', () => {
  let wrapper;

  beforeEach(() => {
   wrapper = shallow(<App />);
   wrapper.setState({ gems: mockGem,
                    metals: mockMetal,
                    selectedGem:{},
                    selectedMetal: {},
                    display: 'gems'})
 })

  it('should match the snapshot with mock state', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with metals page state', () => {
    wrapper.setState({display: 'metals'})
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with results page state', () => {
    wrapper.setState({display: 'results'})
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should select a gem object for state',() => {
    wrapper.instance().selectGem(1)
    expect(wrapper.state('selectedGem')).toEqual(mockGem[0])
  });

  it('should select a metal object for state',() => {
    wrapper.instance().selectMetal(1)
    expect(wrapper.state('selectedMetal')).toEqual(mockMetal[0])
  });

});
