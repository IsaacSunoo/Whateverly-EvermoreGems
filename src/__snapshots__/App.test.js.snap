// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App initial state should match the snapshot with initial state 1`] = `
<div>
  <GemsPage
    gems={Array []}
    selectGem={[Function]}
  />
</div>
`;

exports[`App should match the snapshot with metals page state 1`] = `
<div>
  <MetalsPage
    metals={
      Array [
        Object {
          "id": 1,
          "name": "Gold",
        },
        Object {
          "id": 2,
          "name": "Silver",
        },
      ]
    }
    selectMetal={[Function]}
  />
</div>
`;

exports[`App should match the snapshot with metals page state 2`] = `
<div>
  <ResultsPage
    selectedGem={Object {}}
    selectedMetal={Object {}}
  />
</div>
`;

exports[`App should match the snapshot with mock state 1`] = `
<div>
  <GemsPage
    gems={
      Array [
        Object {
          "id": 1,
          "name": "Aquamarine",
        },
        Object {
          "id": 2,
          "name": "Bixbite",
        },
      ]
    }
    selectGem={[Function]}
  />
</div>
`;
